test/integ/axpy/src/axpy.cpp

```diff
@@ -166,10 +166,9 @@ TEMPLATE_LIST_TEST_CASE("axpy", "[axpy]", TestAccs)
     auto const taskKernel = alpaka::createTaskKernel<Acc>(
         workDiv,
         kernel,
-        numElements,
         alpha,
-        alpaka::getPtrNative(memBufAccX),
-        alpaka::getPtrNative(memBufAccY));
+        alpaka::experimental::readAccess(memBufAccX),
```

Comment:
clang complains about this when in C++17 mode:

```
/usr/lib64/gcc/x86_64-pc-linux-gnu/11.1.0/../../../../include/c++/11.1.0/type_traits:2933:5: error: no type named 'type' in 'std::invoke_result<AxpyKernel, const alpaka::AccCpuSerial<std::integral_constant<unsigned long, 1>, unsigned long> &, const unsigned long &, const float &, const alpaka::experimental::Accessor<float *, float, unsigned long, 1, alpaka::experimental::ReadAccess> &, const alpaka::experimental::Accessor<float *, float, unsigned long, 1, alpaka::experimental::ReadWriteAccess> &>'
    using invoke_result_t = typename invoke_result<_Fn, _Args...>::type;
    ^~~~~
```

The accessors created here are passed by reference to the lower layers but the kernel (correctly) expects them to be passed by value.

Reply:
I can try to have a look later.

Reply:
I can build all tests and examples with clang-12 locally now.

Reply:
DPCPP appears to be based on clang-13. Maybe something changed between clang versions?

Reply:
I guess we need to get #1485 in then.

Reply:
I think I found the culprit:

https://github.com/alpaka-group/alpaka/blob/f9b1b7f7b37954288ad08ad905fa577ee1776c42/include/alpaka/kernel/Traits.hpp#L208-L218

Note that all arguments are transformed into a `const` reference during the check.

No idea why this works for other compilers, though.

Reply:
Never mind. It turns out that I'm stupid and forgot to delete the `numElements` parameter from the kernel instantiation call. Everything works as expected now.